System/Atom.cpp

```diff
@@ -890,15 +891,28 @@ namespace RTE {
 						// Undo scene wrapping, if necessary
 						g_SceneMan.WrapPosition(intPos[X], intPos[Y]);
 
-						// TODO: improve sticky logic!
 						// Check if particle is sticky and should adhere to where it collided
-						if (m_Material->GetStickiness() >= RandomNum() && velocity.MagnitudeIsGreaterThan(0.5F)) {
-							// SPLAT, so update position, apply to terrain and delete, and stop traveling
-							m_OwnerMO->SetPos(Vector(intPos[X], intPos[Y]));
-							m_OwnerMO->DrawToTerrain(g_SceneMan.GetTerrain());
-							m_OwnerMO->SetToDelete(true);
-							m_LastHit.Terminate[HITOR] = hit[dom] = hit[sub] = true;
-							break;
+						if (velocity.MagnitudeIsGreaterThan(1.0F)) {
+							if (m_Material->GetStickiness() >= RandomNum()) {
```

Comment:
I wonder if  two seperate ifs, not an else-if chain, might be better, but to be honest this still looks okay. two seperate ifs would allow a particle to both stick *and* stain, which isn't necessarily an issue. Otherwise as an else-if there's a strange interaction in probability here.

Not particularly fussed though, this looks pretty good either way